3141- fixing the why section to include more descriptive (#3152)
treeverse/lakeFS: docs/index.md

## Why use lakeFS? and what it can do

With lakeFS data teams can stop worrying about issues that take away much of their time:

> * Roll back changes to data: recover quickly from bugs and mistakes directly on your production data.

Reviewer: I think the "directly" here is a bit scary.

> * Test and validate your data before propagating it to data consumers.

Reviewer 2: ```suggestion
* Test and validate your data before propagating it to consumers.
```

> * Travel back in time, across data versions and across data collections - for reproducing and debugging data issues.

Reviewer 2: Maybe put the value first?

```suggestion
* Reproduce and debug data issues by traveling back in time, across data versions and across data collections.
```

> Since lakeFS is compatible with the S3, Azure Blog and GCP object storage API, all popular applications will work without modification, by simply adding the branch name to the object path:

Reviewer: Unfortunately this is not true - while we do support all cloud platforms as the underlying storage - our API is compatible with S3 but not with the other two.

Author: thanks, updating!
